TechDocs: add note to docs about limitations to the scaffolder (#2134)
spotify/backstage: docs/features/techdocs/creating-and-publishing.md

Create an entity from the documentation template and you will get the needed
setup for free.

> !!! warning Currently the scaffolder plugin is limited to create repositories

Reviewer: Please refer to this as the "Backstage Software Templates" since it is the official name of this feature.

> !!! warning Currently the scaffolder plugin is limited to create repositories

Author: ```suggestion
!!! warning Currently the Backstage Software Templates are limited to create repositories
```
